spec/features/admin/dashboard/index_spec.rb

```diff
@@ -1,31 +1,105 @@
+  end
+
+  describe 'it has an incomplete invoices section' do
+    before :each do
+      @customer_1 = FactoryBot.create(:customer)
+      @customer_2 = FactoryBot.create(:customer)
+      @invoice_1 = FactoryBot.create(:invoice, customer: @customer_1)
+      @invoice_2 = FactoryBot.create(:invoice, customer: @customer_1)
+      @invoice_3 = FactoryBot.create(:invoice, customer: @customer_2)
+      @invoice_4 = FactoryBot.create(:invoice, customer: @customer_2)
+
+      @merchant_1 = FactoryBot.create(:merchant)
+      @item_1 = FactoryBot.create(:item, merchant: @merchant_1)
+      @item_2 = FactoryBot.create(:item, merchant: @merchant_1)
+
+      @invoice_item_1 = FactoryBot.create(:invoice_item, item: @item_1, invoice: @invoice_1, status: 'pending')
+      @invoice_item_2 = FactoryBot.create(:invoice_item, item: @item_1, invoice: @invoice_2, status: 'pending')
+      @invoice_item_2 = FactoryBot.create(:invoice_item, item: @item_2, invoice: @invoice_2, status: 'shipped')
+      @invoice_item_3 = FactoryBot.create(:invoice_item, item: @item_1, invoice: @invoice_3, status: 'shipped')
+      @invoice_item_4 = FactoryBot.create(:invoice_item, item: @item_2, invoice: @invoice_4, status: 'packaged')
+    end
+
+    it 'lists all incomplete invoices' do
+      visit '/admin/dashboard'
+
+      within("#incomplete_invoices") do
+        expect(page).to have_content("Incomplete Invoices")
+        expect(page).to have_content(@invoice_1.id)
+        expect(page).to have_content(@invoice_2.id)
+        expect(page).to have_content(@invoice_4.id)
+        expect(page).to_not have_content(@invoice_3.id)
+      end
+    end
+
+    it 'orders the invoices from oldest to newest
+    and displays invoice created at date like "Monday, July 18,2019"' do
+      visit '/admin/dashboard'
+
+      within("#incomplete_invoices") do
+        expect(page).to have_content("Incomplete Invoices")
+        expect("#{@invoice_4.id}").to appear_before("#{@invoice_2.id}")
+        expect("#{@invoice_2.id}").to appear_before("#{@invoice_1.id}")
+
+        within("#incomplete_invoice-#{@invoice_4.id}") do
+          expect(page).to have_content("#{@invoice_4.id}")
+          expect(page).to have_content("#{@invoice_4.created_at.strftime("%A, %B %d, %Y")}")
+        end
+
+        within("#incomplete_invoice-#{@invoice_2.id}") do
+          expect(page).to have_content("#{@invoice_2.id}")
+          expect(page).to have_content("#{@invoice_2.created_at.strftime("%A, %B %d, %Y")}")
+        end
+      end
+    end
+
+    it 'each incomplete invoice is a link to that invoices show page' do
+      visit '/admin/dashboard'
+
+      within("#incomplete_invoice-#{@invoice_4.id}") do
+        expect(page).to have_content("#{@invoice_4.created_at.strftime("%A, %B %d, %Y")}")
+        expect(page).to have_link("#{@invoice_4.id}")
+        click_link("#{@invoice_4.id}")
+        expect(current_path).to eq("/admin/invoices/#{@invoice_4.id}")
+      end
+
+      visit '/admin/dashboard'
```

Comment:
I would suggest using the path helper `admin_dashboard_path` and place these inside a `before :each` hook.

Reply:
I understand the reasoning to help dry up, and I see how you appliedd it in the admin/merchants/index.  I applied that change and removed the original 'visit "/admin/dashboard"' and 3 failures came up.  I'm ok with leaving it as is.